src/core/shelf.sol

```diff
@@ -80,54 +91,105 @@ contract Shelf is TitleOwned {
         shelf[loan].tokenId = nft_;
     }
 
-    function file(uint loan, address registry_, uint nft_, uint principal_) public auth {
-        shelf[loan].registry = registry_;
-        shelf[loan].tokenId = nft_;
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    function token(uint loan) public view returns (address registry, uint nft) {
+        return (shelf[loan].registry, shelf[loan].tokenId);
     }
 
-    function file(uint loan, uint principal_) public auth {
-        shelf[loan].principal = principal_;
-        shelf[loan].initial = principal_;
+    // --- Shelf: Loan actions ---
+    function issue(address registry, uint token) public returns (uint) {
```

Comment:
Currently, the method is public. I don't know if we should have an auth here with a wrapper contract which allows every address to pass but we have the option to block it.

Everybody could just spam our loan issue method.  (Which might be a problem in the UI)

Reply:
@manuelpolzhofer that was intentional. If we make it an auth method we have to add the admit functionality back and it doesn't give us a lot of benefits. I don't think we have to worry about the UI at the smart contract level :)